common/src/main/java/com/microsoft/identity/common/AndroidPlatformComponents.java

```diff
@@ -110,7 +111,12 @@ private static synchronized void initializeStaticClasses(@NonNull final Context
         if (!sInitialized) {
             Device.setDeviceMetadata(new AndroidDeviceMetadata());
             Logger.setAndroidLogger();
-            HttpCache.initialize(context.getCacheDir());
+            final File cacheDir = context.getCacheDir();
+            if (cacheDir != null) {
+                HttpCache.initialize(cacheDir);
+            } else {
+                Logger.warn(TAG, "Http caching is not enabled because the cache dir is null");
```

Comment:
Can this happen in production ? and should this be an error in that case?

Reply:
I'm unsure.  We **should not** depend on http caching for anything other than performance...